CONTRIBUTING.md

```diff
@@ -1,196 +1,343 @@
+Links to Drone builds results can be found within a pull request and on the list
+of changes pushed to a branch. Results can also be browsed [directly][dronevic].
 
-If you don't have a running ESX required for tests, you can leverage the automated Drone servers for
-running tests. Add `WIP` (work in progress) to the PR title to alert reviewers that the PR is not ready to be merged.
+As the Drone environment is a shared resource, it is best to run tests locally
+before publishing a pull request. If you don't have an environment suitable for
+running the tests, you may leverage the Drone environment. When doing so, it is
+best to include `[WIP]` (work in progress) at the beginning of the title of the
+pull request to alert readers that the change is not ready for review.
 
-If your Drone build needs to be restarted, fork the build:
+Drone builds can be restarted via the web interface or the [CLI][dronecli]:
 ```shell
-export DRONE_TOKEN=<Drone Token>
+export DRONE_TOKEN=<Drone Token, from https://ci-vic.vmware.com/account/token>
 export DRONE_SERVER=https://ci-vic.vmware.com
 
 drone build start vmware/vic <Build Number>
 ```
-If you are not a member of `vmware` org in github, then your PR build may fail. In that case, request one of the existing members / reviewers to fork your failed build to skip membership checking.
+
+For security reasons, your pull request build may fail if you are not a member
+of the `vmware` organization in GitHub. If this occurs, leave a comment on your
+pull request asking that the build be restarted by an organization member.
+
+To skip membership checking when restarting a build:
 ```shell
 drone build start --param SKIP_CHECK_MEMBERSHIP=true vmware/vic <Build Number>
 ```
 
-### Testing locally
+[dronevic]:https://ci-vic.vmware.com/vmware/vic
+[dronesrc]:https://github.com/drone/drone
+[dronecli]:http://docs.drone.io/cli-installation/
 
-Developers need to install [Drone CLI][dronecli].
 
-#### Unit tests
+## Reporting bugs and creating issues
 
-``` shell
-VIC_ESX_TEST_URL="<USER>:<PASS>@<ESX IP>" drone exec .drone.yml
-```
+Communicating clearly helps with efficient triage and resolution of reported
+issues.
 
-If you don't have a running ESX, tests requiring an ESX can be skipped with the following:
+The summary of each issue will likely be read by many people. Quickly conveying
+the essence of the problem you are experiencing helps get the right people
+involved. Reports which are vague or unclear may take longer to be routed to
+a domain expert.
 
-``` shell
-drone exec
+The body of an issue should communicate what you are trying to accomplish and
+why; understanding your goal allows others to suggest potential workarounds. It
+should include specific details about what is (or isn't happening).
+
+Proactively including screenshots and logs can be very helpful. When including
+logs, please ensure that formatting is preserved by using [code blocks][code].
```

Comment:
When including log snippets inline in the PR instead of as an attachment, please ensure ...

Maybe we need a link to how to collect container log bundles - and possibly a note about using debug=1 if the issue can be recreated.